roles/rsyslog/tasks/main.yml

```diff
@@ -64,25 +64,26 @@
         remove: '{{ true if rsyslog_purge_original_conf | bool else false }}'
       changed_when: false
 
-    - name: Get directory stat of rsyslog_system_log_dir (/var/log)
+    - name: Get directory stat of rsyslog_system_log_dir (/var/log, by default)
       stat:
         path: "{{ rsyslog_system_log_dir }}"
       register: __rsyslog_system_log_dir_stat
 
-    - name: Update directory and file permissions
+    - name: Create directory if it does not exist or is not a directory.
       file:
         state: directory
         path: '{{ rsyslog_system_log_dir }}'
         mode: '0700'
-      when: (not __rsyslog_system_log_dir_stat.stat.isdir) or
-            (__rsyslog_system_log_dir_stat.stat.uid != 0 or __rsyslog_system_log_dir_stat.stat.gid != 0) or
-            (__rsyslog_system_log_dir_stat.stat.mode != '0700')
+      when: ((not __rsyslog_system_log_dir_stat.stat.exists) or
+             (not __rsyslog_system_log_dir_stat.stat.isdir) or
+             (__rsyslog_system_log_dir_stat.stat.uid != 0) or
+             (__rsyslog_system_log_dir_stat.stat.gid != 0) or
+             (__rsyslog_system_log_dir_stat.stat.mode != '0700'))
```

Comment:
The `file` module is supposed to do all of this `when` stuff for you.  AFAICT, you can replace all of the `when` conditions with
```yaml
file:
  state: directory
  path: '{{ rsyslog_system_log_dir }}'
  mode: '0700'
  owner: root
  group: root
```

Reply:
and then you can get rid of the `stat`, if you don't need the result elsewhere.

Reply:
One question.  Do we care the path already exists as a file, the ansible-playbook fails with this message.  It fails with my original patch, too.  :p  So, my question would be if the path exists not as a directory, do we want to remove it prior to the task or it's ok to fail?
```
    "mode": "0666",
    "msg": "/tmp/var/log already exists as a file",
    "owner": "nhosoi",
    "path": "/tmp/var/log",
```

Reply:
> One question. Do we care the path already exists as a file,

No, I don't think so.  If we do, there are probably several dozen places in other roles where we would need to put such a check.

> the ansible-playbook fails with this message. It fails with my original patch, too. :p So, my question would be if the path exists not as a directory, do we want to remove it prior to the task or it's ok to fail?

I think it is ok to fail.

> 
> ```
>     "mode": "0666",
>     "msg": "/tmp/var/log already exists as a file",
>     "owner": "nhosoi",
>     "path": "/tmp/var/log",
> ```

Reply:
Thank you, @richm and @pcahyna.  Pushed the fix.